### PR TITLE
[FSDP2] privateuse1 support fsdp2.

### PR DESCRIPTION
**Pull request by will-core**
We are looking forward to supporting FSDP2 with devices other than CUDA. Please give me some coding suggestions. Thank you very much.

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: torch/distributed/_composable/fsdp/_fsdp_param_group.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ class FSDPCommContext:
 
     def lazy_init(self, device: torch.device):
         self.device_handle = _get_device_handle(device.type)
-        if device.type not in ["cuda", "hpu"]:
+        if device.type not in ["cuda", "hpu", torch._C_get_privateuse1_backend_name()]:
```
Comment (kwen2501) [inline torch/distributed/_composable/fsdp/_fsdp_param_group.py:54]: Should `torch._C_get_privateuse1_backend_name` be
`torch._C._get_privateuse1_backend_name`?
(missing one `.`)

**Review by kwen2501 — COMMENTED** [review]
@awgu thinking out loud -- how about we just remove the device type check here and let the program fail at the place where it needs to access stream? I believe / hope there would be equivalent error message from core that would let user know the same thing.

**Review by kwen2501 — APPROVED** [review]
CI looks good. Approving.

**Comment by will-core** [comment]
Please take a look at this PR and see if there are any other suggestions. Thank you. @awgu @weifengpy

**Comment by awgu** [comment]
yea it's probably fine
I do not think this is a major concern

**Pull request closed without merging**